### ISSUE TITLE
glfwGetKeyName creates GLFW_INVALID_VALUE error

### ISSUE THREAD

**Issue by racz16** [glfw/glfw#2214]
If I press the mute key (or any other special key) on my keyboard, I get `GLFW_KEY_UNKNOWN` as the `key`, and `256` as the `scancode` in `glfwSetKeyCallback`. If I pass these parameters to the function `glfwGetKeyName`, I get an error in the function `glfwSetErrorCallback`, where the `error` is `GLFW_INVALID_VALUE` and the `description` is `Invalid scancode 256`. 
However, the documentation states for the function `glfwGetKeyName`, that
> If the key is `GLFW_KEY_UNKNOWN`, the scancode is used to identify the key, otherwise the scancode is ignored. If you specify a non-printable key, or `GLFW_KEY_UNKNOWN`, and a scancode that maps to a non-printable key, this function returns `NULL` but does not emit an error.

and in the Errors section, it states that
> Possible errors include `GLFW_NOT_INITIALIZED` and `GLFW_PLATFORM_ERROR`.

I think the behavior described in the documentation is preferred because if GLFW can't determine the name of the key, it should return a `nullptr` and let me handle the situation instead of creating an error because I can't tell before calling the function that it will find the name or not.
I use Windows 11 22H2, with hungarian keyboard layout, the hungarian version of the Logitech Prodigy G213 keyboard, and GLFW 3.3.8.

**Comment by dougbinks** [comment]
This is an issue for many apps, for example the popular [bloat-free graphical user interface library for C++, Dear ImGui, has this issue](https://github.com/ocornut/imgui/issues/5908), and they use a workaround to disable and eat any errors. I use a similar workaround in my programs.

I do not think valid data from end user systems should be considered an error, so I would recommend removing the line:

```C
        _glfwInputError(GLFW_INVALID_VALUE, "Invalid scancode %i", scancode);
```

from the platform `_glfwGetScancodeName*` functions. I can do this work and submit a PR if needed.

**Comment by elmindreda** [comment]
This was indeed a bug in the implementation and should be fixed now with 86bf5698ecfdbe783135ffaf72972a8c562c0fdf, which will be included in 3.3.10 and 3.4. Thank you for the bug report!
